Mining Vendor/Interdyne Balance Changes

Thlumyn opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Some changes to make things easier on Interdyne:

Added some mining items to the mining vendor (for both station and Interdyne) that are printable on the station, but not previously obtainable on Interdyne:
- Advanced Plasma Cutter
- Mining AoE PKA attachment
- Trigger Guard PKA attachment
- Night Vision Mesons
- Prescription Mesons
- Mining Satchel of Holding (this is obtainable through the frontier rapid construction fabricator as well)
- Night Vision Health Meson HUD (though this was NOT available on station either, before)

This also adds another deluxe stock parts box to Interdyne, and makes the Slime Processor, Monkey Recycler and ChemMaster start fully upgraded. 

This adds an advanced health analyzer on the table next to the advanced surgery duffelbag.

And it puts a handheld recoverer in their xenoarch locker, because I'm just too stupid to figure it out, but I can't for the life of me get the items recovered in xenoarch otherwise

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## How This Contributes To The Nova Sector Roleplay Experience

With the new changes to Interdyne, Interdyne player count seems to have gone down a bit, especially on the mining side of things. While the idea was to have Interdyne trade with DS-2 or the station, that doesn't work out all the time, in my opinion. Quite often there is nobody active on DS-2. Because of the limited stock parts available, quite often the first people to spawn just use it on their own section (eg, chemistry, botany, xenobiology) which caused some friction between players. Hopefully this will be alleviated by making some of the pharmaceutical adjacent machines tier 4 at round start, and with an extra box to upgrade some of the other round start machines. This still does not cover all the machines, and if Interdyne wants to build new things, they will still need to get additional parts/boards from the station/DS-2 - most of the trade that I've seen with the station/DS-2 about parts/machines before have been this case anyway, where Interdyne has been trying to get boards such as getting the frontier boards from station cargo. Some Interdyne items already start upgraded, such as the drink machines and chem machine, so I don't think this is too far off.

About the additional items in the mining vendor: while we can print mining satchel of holdings from the frontier fab, some of the other items that make mining more pleasant were only printable at the cargo fab. This allows the station to print those items from the mining vendor as well, though at a mining points price, so hopefully there is still incentive for the station people to wait for research to get those items instead. As for the night vision health scanner hud: that is just so nice to have while fighting fauna and mining, I always use it on my local server when I practice megafauna fighting, and I think it would just be a nice thing for people to be able to buy from the vendor. Getting materials is among one of the first mechanical processes that people need to get through to set up Interdyne, and considering that we can't really print things without it, not able to effectively mine too well until going through the process of requesting boarding from NT, waiting for them to research, and trying to get the departments to give parts - which can be hard even as a station character - just made mining and waiting for materials at Interdyne less pleasant. 

<!-- Please add a short description of why you think these changes would benefit the game and the roleplay atmosphere of the server. If you can't justify it in words, it might not be worth adding. -->

## Proof of Testing

<!-- Include any screenshots/videos/debugging steps of the code functioning successfully, between the </summary> and </details> code blocks. -->
<!-- To our mappers and spriters: Posting screenshots of content INSIDE EDITORS (aseprite, PDN, SDMM, ect) is NOT valid proof of testing. Please make sure that you COMPILE the game and provide PROOF you tested your edits. -->

<details>
<summary>Screenshots/Videos</summary>
  
![image](https://github.com/NovaSector/NovaSector/assets/102194057/05394279-85bf-4abe-8778-3e3205a79b38)
![image](https://github.com/NovaSector/NovaSector/assets/102194057/3f9b3d03-f32e-4fe2-8cb3-527dafc1509c)
![image](https://github.com/NovaSector/NovaSector/assets/102194057/eb496099-c04d-4596-babf-faf672752182)

</details>

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: added some cargo lathe items to the mining equipment vendor
balance: add night vision health scanner mesons to the mining equipment vendor 
balance: change some Interdyne machines to start at tier 4
balance: gives Interdyne another box of upgraded parts
balance: gives Interdyne a handheld recoverer
balance: gives Interdyne an advanced health analyzer
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

sqnztb commented:
Night vision health huds and mesons are already available on interdyne - 
![image](https://github.com/NovaSector/NovaSector/assets/28007787/d51eb183-4d41-4e42-a12f-11ec4f322905)

Making the ore satchels buyable is also kind of weird since the RCF just needs gold and uranium to print them, which is already available in the virology lab.

I would argue that making the chemmaster fully upgraded is... kinda pointless considering you start with a box of bluespaces on the desk and servos just make you print faster. Putting another box of deluxe parts AND upgrading the few things that you'd normally use the stock parts on is... also questionable.

I'm also genuinely confused about why you nuked the 
`modular_nova/modules/mapping/code/interdyne_mining.dm`
just to roll it into
`modular_nova/modules/mapping/code/interdyne.dm` 
... which again seems unnecessary.

None of this is really "balance", this is just wanting more free stuff at the start of the round, none of which is necessary.

Thlumyn commented:
I've never seen the xenobiology stuff upgraded unless I'm the one to get there first and upgrade it; usually the parts are all spent on other things, such as the medical lathe, recycler, autolathe, ORM, chemmaster. By the time the medical lathe and ORM are built, there's only 3 servos left, and a lot more than 3 machines that require servos. I can see removing the box, _or_ not having those start upgraded, but 2 boxes for all they have now and not is just not enough. 

I changed the file name because originally the code was just related to interdyne mining stuff, but with those other machine types I added, it wasn't all mining, so removed the mining from the file name. 

I'd be fine removing the night vision mesons and satchel from the mining vendor

Thlumyn commented:
Closing this as I'm taking a break from Interdyne anyway.

Thlumyn commented:
@sqnztb 

Okay, I removed the box of parts, and changed the biogenerator to be fully upgraded. 
I also removed the handheld recoverer
I removed the mining satchel, night vision mesons and trigger guard from the vendor 

I also changed the "Interdyne Bioweapon Scientist" name to the "Interdyne Scientist". This is to make them seem less openly hostile, especially considering that they're now meant to be going up to the station and trading and _probably_ shouldn't be opening telling the station they make bioweapons, but those are agent IDs so if people really want that name, they can change themselves.

GoldenAlpharex commented:
I think the mining vendor changes should be unique to Interdyne only, as they kind of circumvent basically the need for miners to wait for Science basically entirely.

Thlumyn commented:
> I think the mining vendor changes should be unique to Interdyne only, as they kind of circumvent basically the need for miners to wait for Science basically entirely.

I made the mining vendor changes Interdyne only 
![image](https://github.com/NovaSector/NovaSector/assets/102194057/579be709-99fd-4cc6-9c23-b27c61a8fad0)
